sources/api/certdog/src/main.rs

```diff
@@ -0,0 +1,355 @@
+
+    // SimpleLogger will send errors to stderr and anything less to stdout.
+    SimpleLogger::init(args.log_level, LogConfig::default()).context(error::Logger)?;
+
+    info!("certdog started");
+    let certificate_bundles = get_certificate_bundles(&args.socket_path).await?;
+    info!("Got certificate bundles from API");
+    update_trusted_store(certificate_bundles, args.trusted_store, args.source_bundle)?;
+    info!("Updated trusted store");
+
+    Ok(())
+}
+
+// Returning a Result from main makes it print a Debug representation of the error, but with Snafu
+// we have nice Display representations of the error, so we wrap "main" (run) and print any error.
+// https://github.com/shepmaster/snafu/issues/110
+#[tokio::main]
+async fn main() {
+    if let Err(e) = run().await {
+        error!("{}", e);
+        process::exit(1);
+    }
+}
+
+// =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+mod error {
+    use snafu::Snafu;
+    use std::io;
+    use std::path::PathBuf;
+
+    #[derive(Debug, Snafu)]
+    #[snafu(visibility = "pub(super)")]
+    pub(super) enum Error {
+        #[snafu(display("Error sending {} to {}: {}", method, uri, source))]
+        APIRequest {
+            method: String,
+            uri: String,
+            source: apiclient::Error,
+        },
+
+        #[snafu(display("Unable to decode base64 from certificate '{}': '{}'", name, source))]
+        Base64Decode {
+            name: String,
+            source: base64::DecodeError,
+        },
+
+        #[snafu(display("Logger setup error: {}", source))]
+        Logger { source: log::SetLoggerError },
+
+        #[snafu(display("Invalid log level '{}'", log_level))]
+        LogLevel {
+            log_level: String,
+            source: log::ParseLevelError,
+        },
+
+        #[snafu(display("Failed to parse PEM: {}", source))]
+        ParsePEM {
+            source: x509_parser::error::PEMError,
+        },
+
+        #[snafu(display("Failed to parse cert: {}", source))]
+        ParseX509Certificate {
+            source: x509_parser::nom::Err<x509_parser::error::X509Error>,
+        },
+
+        #[snafu(display("Error while reading bundle from file {}: '{}'", path.display(), source))]
+        ReadSourceBundle { path: PathBuf, source: io::Error },
+
+        #[snafu(display("Error deserializing response from '{}': '{}'", uri, source))]
+        ResponseJson {
+            uri: String,
+            source: serde_json::Error,
+        },
+
+        #[snafu(display("Failed to update trust store: {}", source))]
+        UpdateTrustedStore { source: io::Error },
+
+        #[snafu(display("Failed to write to pem string: {}", source))]
+        WritePEMString { source: std::fmt::Error },
+    }
+}
+
+type Result<T> = std::result::Result<T, error::Error>;
+
+#[cfg(test)]
+mod test_certdog {
+    use model;
+    use model::modeled_types::{Identifier, PemCertificateString};
```

Comment:
Unless there's a name issue I don't know about, probably cleaner to import `PemCertificate` and drop the import of `model`.

```suggestion
    use model::modeled_types::{Identifier, PemCertificate, PemCertificateString};
```

Reply:
PemCertificate isn't a modeled type, it is a model, and in other crates, we don't include `model` for some reason, we  always do `model::TheModel`

Reply:
OK. 🤷